development/components/database/objectmodel.md

```diff
@@ -64,86 +104,443 @@ public static $definition = [
 ];
 ```
 
-## Multiple stores and/or languages
+Let's analyse this definition :
+
+```php
+public static $definition = [
+    'table' => 'cms',
+    'primary' => 'id_cms',
+    'multilang' => true,
+    'fields' => array(
+```
+
+- `table` is the related database table name (without the database table `PREFIX`, usually `ps_`), 
+- `primary` is the name of the `PRIMARY KEY` field in the database table, which will be used as `$id` in the ObjectModel
+- `multilang` is a boolean value indicating that the entity is available in multiple langages, see [Multiple languages]({{< ref "#multiple-languages" >}})
+- `fields` is an array containing all other of the fields from the database table.
+
+### Fields description
+
+A field is defined by a key (its name in the database table) and an array of description of its settings. 
+
+```php
+'meta_description' => [
+    'type' => self::TYPE_STRING,
+    'lang' => true,
+    'validate' => 'isGenericName',
+    'size' => 255
+],
+```
+
+In this example :
+
+- `meta_description` is the field's name
+- `self::TYPE_STRING` is its type
+- `lang` is a boolean related to multilang features ([Multiple languages]({{< ref "#multiple-languages" >}}))
+- `validate` is a validation rule (optional)
+- `size` is the max size of the field. It should match the database definition to avoid data cropping or overflow when inserting/updating objects.
+
+#### Field types reference
+
+Field type is an important setting, it determines how ObjectModel will format your data.
+
+<table>
+    <thead>
+        <tr>
+            <th>Type in ObjectModel</th>
+            <th>Related type in MySQL</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>TYPE_INT</td>
+            <td>INTEGER, INT, SMALLINT, TINYINT, MEDIUMINT, BIGINT, ...</td>
+        </tr>
+        <tr>
+            <td>TYPE_BOOL</td>
+            <td>SMALLINT</td>
+        </tr>
+        <tr>
+            <td>TYPE_STRING</td>
+            <td>VARCHAR</td>
+        </tr>
+        <tr>
+            <td>TYPE_FLOAT</td>
+            <td>FLOAT, DOUBLE</td>
+        </tr>
+        <tr>
+            <td>TYPE_DATE</td>
+            <td>DATE</td>
+        </tr>
+        <tr>
+            <td>TYPE_HTML</td>
+            <td>BLOB, TEXT</td>
+        </tr>
+        <tr>
+            <td>TYPE_NOTHING</td>
+            <td>Should not be used (avoids ObjectModel formating)</td>
```

Comment:
I wouldn't say, "it should not be used". I'd go for something that is probably not secure and use it cautiously.